src/ipfs/schemas.ts

```diff
@@ -0,0 +1,101 @@
+import { z } from 'zod';
+
+const VERSION_1 = '1.0' as const;
+const VERSION_2 = '2.0' as const;
+
+// Ethereum address regex (case insensitive)
+const ethereumAddressRegex = /^0x[a-f0-9]{40}$/i;
+
+export const coverValidatorsSchema = z.object({
+  version: z.literal(VERSION_1),
+  validators: z
+    .array(z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'))
+    .min(1, 'At least one validator address is required'),
+});
+
+export const coverQuotaShareSchema = z.object({
+  version: z.literal(VERSION_1),
+  quotaShare: z.number().min(0).max(100),
+});
+
+export const coverAumCoverAmountPercentageSchema = z.object({
+  version: z.literal(VERSION_1),
+  aumCoverAmountPercentage: z.number().min(0).max(100),
+});
+
+export const coverWalletAddressSchema = z.object({
+  version: z.literal(VERSION_1),
+  walletAddress: z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'),
+});
+
+export const coverWalletAddressesSchema = z.discriminatedUnion('version', [
+  z.object({
+    version: z.literal(VERSION_1),
+    walletAddresses: z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'),
```

Comment:
version one should allow comma separated addresses as a single string

Reply:
done